versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksRef.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.rocksdb;
+
+import static com.dremio.nessie.versioned.store.rocksdb.Function.EQUALS;
+import static com.dremio.nessie.versioned.store.rocksdb.Function.SIZE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import com.dremio.nessie.tiered.builder.Ref;
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.impl.condition.ExpressionPath;
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.StoreException;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+class RocksRef extends RocksBaseValue<Ref> implements Ref, Evaluator {
+
+  static final String TYPE = "type";
+  static final String NAME = "name";
+  static final String METADATA = "metadata";
+  static final String COMMITS = "commits";
+  static final String COMMIT = "commit";
+  static final String CHILDREN = "children";
+
+  private RefType type;
+  private String name;
+  private Id commit;
+  private Id metadata;
+  private Stream<Id> children;
+  private List<ValueProtos.Commit> commits;
+
+  RocksRef() {
+    super();
+  }
+
+  @Override
+  public boolean evaluate(Condition condition) {
+    // Retrieve entity at function.path
+    if (type == RefType.BRANCH) {
```

Comment:
Why type here, but on line 58 it's this.type?

Reply:
Hmm. Inconsistency. I removed the this on line 58.